cranelift-codegen/meta/src/gen_inst.rs

```diff
@@ -1060,36 +1062,34 @@ fn gen_builder(instructions: &AllInstructions, formats: &FormatRegistry, fmt: &m
         for inst in instructions.values() {
             gen_inst_builder(inst, &*inst.format, fmt);
         }
-        for format in formats.iter() {
+        for format in formats {
             gen_format_constructor(format, fmt);
         }
     });
     fmt.line("}");
 }
 
 pub(crate) fn generate(
-    shared_defs: &SharedDefinitions,
+    formats: &Vec<&InstructionFormat>,
```

Comment:
I would expect `Vec<InstructionFormat>` here?

Reply:
(Ah, I see that we are referring to the `Rc<InstructionFormat>` as a `&InstructionFormat` but what about the `Vec`?)

Reply:
Indeed, it could consume the vector instead of letting it live after the call. Since there's no subsequent uses of this vector, I'll do this.